DigitalLearningSolutions.Data/Services/CourseService.cs

```diff
@@ -159,5 +168,18 @@ public void UpdateCourseOptions(CourseOptions courseOptions, int customisationId
 
             return new DelegateCourseDetails(info, customPrompts, attemptStats);
         }
+
+        public bool RemoveDelegateFromCourseIfDelegateHasCurrentProgress(int delegateId, int customisationId, RemovalMethod removalMethod)
+        {
+            var currentProgress = progressDataService.GetDelegateProgressForCourse(delegateId, customisationId)
+                .FirstOrDefault(p => p.Completed == null && p.RemovedDate == null);

```

Comment:
The GroupService is the only other place where ProgressDataService.GetDelegateProgressForCourse is currently used, but in there we do not use .FirstOrDefault() despite the fact that there should only ever be one record that matches these requirements. I think the logic was that the database does not actually enforce that, therefore we should err on the cautious side and replace the FirstOrDefault() with Where(). Probably good to be consistent here as well.